### PR TITLE
mailer to admins that channel was deleted

### PR DESCRIPTION
**Pull request by met0ry**
- [x] merged with the latest master branch
- [x] resolved all the conflicts, PR could be merged
- [x] new code has new specs
- [x] for new models - new factories were added too
- [x] unit specs are passed
- [x] code is well-formatted, Ruby, CSS, etc
- [x] screenshots of new pages, features are added
- [x] code passed internal team code review
- [x] no instance variables in partials
- [x] code is well formatted and has spaces!!!

PS: delete your branch in github after merge http://prntscr.com/q80mvq

### CODE REVIEW

File: app/mailers/admin_mailer.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+class AdminMailer < ApplicationMailer
+  default from: 'remote-flow@mail.com'
+
+  def delete_channel_email(channel)
```
Comment (romaslmd) [inline app/mailers/admin_mailer.rb]: avoid name duplication in methods
`AdminMailer.delete_channel_email` -> `AdminMailer` already gives a clue that it's a mailer, and method is to specify event

File: app/mailers/admin_mailer.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+    @channel = channel
+    mail(to: admins_email(channel.organization), subject: 'Channel was deleted')
+  end
+
+  private
+
+  def admins_email(organization)
+    organization.users.admins.pluck(:email)
```
Comment (romaslmd) [inline app/mailers/admin_mailer.rb]: remove this method, it's an inline operation, extra private method is redundant

File: app/models/channel.rb [hunk 3]
```diff
@@ -3,9 +3,17 @@ class Channel < ApplicationRecord
...
+
   validates :name, presence: true,
                    length: { minimum: 2,
                              maximum: 255 }
 
   scope :by_name, -> { order(:name) }
+
+  private
+
+  def send_email_notification
```
Comment (romaslmd) [inline app/models/channel.rb]: unclear method name, `notify_deleted` or something like that would give more meaning for readers

File: spec/mailers/admin_mailer_spec.rb [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+RSpec.describe AdminMailer, type: :mailer do
+  describe 'send notification email' do
+    let!(:admin) { create(:user, :admin, organization: organization) }
+    let!(:organization) { create(:organization) }
+    let!(:channel) { create(:channel, organization: organization) }
+    let!(:mail) { AdminMailer.delete_channel_email(channel) }
+
+    it 'deliver success' do
+      expect { channel.destroy }.to change { ActionMailer::Base.deliveries.count }.by(1)
```
Comment (romaslmd) [inline spec/mailers/admin_mailer_spec.rb]: no context, pls make test structure more explicit to understand what and why code is doing

File: app/models/channel.rb (hunk 3 above)
Comment (igorkasyanchuk) [reply app/models/channel.rb]: yes, rename. As a developer I MUST understand what method is doing by it's name

File: spec/mailers/admin_mailer_spec.rb [hunk 5]
```diff
@@ -0,0 +1,22 @@
+require "rails_helper"
+
+RSpec.describe AdminMailer, type: :mailer do
+  describe 'send notification email' do
+    let!(:admin) { create(:user, :admin, organization: organization) }
+    let!(:organization) { create(:organization) }
+    let!(:channel) { create(:channel, organization: organization) }
+    let!(:mail) { AdminMailer.delete_channel_email(channel) }
+
+    it 'deliver success' do
```
Comment (igorkasyanchuk) [inline spec/mailers/admin_mailer_spec.rb]: this test is in the wrong place
you testing here NOT mailer
you are testing the model that it's triggering sending a notification email.
Move to the model

AND you don't need to have anything like `let!(:mail) { AdminMailer.delete_channel_email(channel) }` there
JUST channel

File: spec/mailers/previews/admin_mailer_preview.rb [hunk 6]
```diff
@@ -0,0 +1,4 @@
+# Preview all emails at http://localhost:3000/rails/mailers/admin_mailer
+class AdminMailerPreview < ActionMailer::Preview
+
```
Comment (igorkasyanchuk) [inline spec/mailers/previews/admin_mailer_preview.rb]: а це нащо?

**Pull request merged**